src/dotVariant.Generator/Diagnose.cs

```diff
@@ -220,7 +226,7 @@ private static class CheckThat
             {
                 foreach (var p in options)
                 {
-                    if (Inspect.IsAncestorOf(p.Type, type))
+                    if (Inspect.IsAncestorOf(p.Type, type) && CheckNoImplicitConversionAttribute(p))
```

Comment:
Please add a new class to [NoImplicitConversionForBaseClasses.cs](https://github.com/mknejp/dotvariant/blob/stable/src/dotVariant.Generator.Test/diagnostics/NoImplicitConversionForBaseClasses.cs) with the attribute present to test that the warning is silenced.

Reply:
Oh nice, didn't even see that you had diagnostic tests